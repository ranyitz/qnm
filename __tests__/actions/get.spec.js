const getAction = require('../../src/actions/get');
const { resolveWorkspace } = require('../utils');

describe('get', () => {
  it('should get the version of a single module', () => {
    const workspace = resolveWorkspace('single-module');
    const output = getAction(workspace, 'test');

    expect(output).toMatchSnapshot();
  });

  it('should show a message when no module has found', () => {
    const workspace = resolveWorkspace('single-module');
    try {
      getAction(workspace, 'not-exist');
    } catch (e) {
      expect(e.message).toMatchSnapshot();
    }
  });

  it('should suggest an alternative when no module has found and there is an alternative with an edit distance smaller than 2', () => {
    const workspace = resolveWorkspace('single-module');
    try {
      getAction(workspace, 'dest');
    } catch (e) {
      expect(e.message).toMatchSnapshot();
      expect(e.message).toMatchSnapshot();
    }
  });

  it('should get the version of a module in depth', () => {
    const workspace = resolveWorkspace('module-in-depth');
    const output = getAction(workspace, 'test');

    expect(output).toMatchSnapshot();
  });

  it('should get the version of a single module when not starting at the root', () => {
    const workspace = resolveWorkspace('single-module/node_modules');
    const output = getAction(workspace, 'test');

    expect(output).toMatchSnapshot();
  });

  it('should get the version of a single module when in a scoped package', () => {
    const workspace = resolveWorkspace('scoped-module');
    const output = getAction(workspace, '@scope/test');

    expect(output).toMatchSnapshot();
  });

  it('should print versions in three levels deep including ancestors', () => {
    const workspace = resolveWorkspace('three-levels-deep');
    const output = getAction(workspace, 'dep-of-dep-of-dep');

    expect(output).toMatchSnapshot();
  });
});
