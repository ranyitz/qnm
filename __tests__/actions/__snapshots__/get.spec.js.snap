// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get should disable colors 1`] = `
"[4mtest[24m
â””â”€â”¬ [90manother[39m
  â””â”€â”€ 1.0.0
"
`;

exports[`get should get the version of a module in depth 1`] = `
"[4mtest[24m
â””â”€â”¬ [90manother[39m
  â””â”€â”€ [37m1.0.0[39m
"
`;

exports[`get should get the version of a single module 1`] = `
"[4mtest[24m
â””â”€â”€ [37m1.0.0[39m
"
`;

exports[`get should get the version of a single module when in a scoped package 1`] = `
"[4m@scope/test[24m
â””â”€â”€ [37m1.0.0[39m
"
`;

exports[`get should get the version of a single module when not starting at the root 1`] = `
"[4mtest[24m
â””â”€â”€ [37m1.0.0[39m
"
`;

exports[`get should print versions in three levels deep including ancestors 1`] = `
"[4mdep-of-dep-of-dep[24m
â””â”€â”¬ [90mdep[39m
  â””â”€â”¬ [90mdep-of-dep[39m
    â””â”€â”€ [37m1.0.0[39m
"
`;

exports[`get should show a message when no module has found 1`] = `"Could not find any module by the name \\"not-exist\\"."`;

exports[`get should suggest an alternative when no module has found and there is an alternative with an edit distance smaller than 2 1`] = `"Could not find any module by the name \\"dest\\". Did you mean \\"[1mt[22mest\\"?"`;

exports[`get should suggest an alternative when no module has found and there is an alternative with an edit distance smaller than 2 2`] = `"Could not find any module by the name \\"dest\\". Did you mean \\"[1mt[22mest\\"?"`;
